ganga/GangaGUI/gui/routes.py

```diff
@@ -64,13 +83,365 @@ def decorated(*args, **kwargs):
     return decorated
 
 
-# TODO resource route for development purpose - REMOVE LATER
-# Protected resource
-@app.route("/resource", methods=["POST"])
+# ******************** Job API ******************** #
+
+# Single Job Info API - GET Method
+@app.route("/job/<int:job_id>", methods=["GET"])
+@token_required
+def job_endpoint(current_user, job_id: int):
+    """
+    Given the job id returns the general information related to the job in JSON format.
+
+    Returned job information: fqid, status, name, number of subjobs, application, backend, backend.actualCE, comments, subjobs statuses.
+
+    :param job_id: int
+    """
+
+    # Imports
+    from GangaCore.GPI import jobs
+
+    # Job existence check
+    if job_id not in list(jobs.ids()):
+        return jsonify({"success": False, "message": "Job with ID {} does not exist".format(job_id)}), 400
+
+    # Get the general job info
+    job_data = get_job_data(job_id)
+
+    return jsonify(job_data)
+
+
+# Single Job Attribute Info API - GET Method
+@app.route("/job/<int:job_id>/<attribute>", methods=["GET"])
+@token_required
+def job_attribute_endpoint(current_user, job_id: int, attribute: str):
+    """
+    Given the job id and attribute, return the attribute information in the string format via JSON.
+
+    Supported attributes: application, backend, do_auto_resubmit, fqid, id, info, inputdir, inputfile, master, name, outputdir, outputfiles, parallel_submit, splitter, status, subjobs, time
+
+    :param job_id: int
+    :param attribute: str
+    """
+
+    # Imports
+    from GangaCore.GPI import jobs
+
+    # Job existence check
+    if job_id not in list(jobs.ids()):
+        return jsonify({"success": False, "message": "Job with ID {} does not exist".format(job_id)}), 400
+
+    supported_attributes = ["application", "backend", "do_auto_resubmit", "fqid", "id", "info", "inputdir", "inputdata",
+                            "inputfiles",
+                            "master", "name", "outputdir", "outputdata",
+                            "outputfiles", "parallel_submit", "splitter", "status", "subjobs", "time"]
```

Comment:
Is this list of information available elsewhere?

Reply:
Actually the list is just to identify what attribute info is the user requesting. But essentially we are taking the data from the actual attribute implementation of the Job only so if any changes are made to the internal working of the attribute in the GangaCore, they will be reflected in the GUI. 

But if a case comes that a new attribute is added to the Job class, then it will be needed to add that in this list for it to be accessible to the GUI.